### DISCUSSION TITLE
Terragrunt init /plan nothing works - Remote state S3 bucket does not exist

### DISCUSSION THREAD

**Discussion by skkc2** [gruntwork-io/knowledge-base#609] [Help]
Hi Team,

I've been facing the below problem while running the terragrunt init, plan and other commands. My terraform code works fine and I've been working with terragrunt never faced such kind of issue. 

When we run terragrunt init, plan etc, Terragrunt errors out as-

>  " Remote state S3 bucket tfstate-devops-123456789-eu-west-1 does not exist or you don't have permissions to access it. Would you like Terragrunt to create it? (y/n) n
> level=error msg=remote state S3 bucket tfstate-devops-123456789-eu-west-1 does not exist or you don't have permissions to access it
> level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1 "

Little more info 

- We're trying to update the lambda function in aws, we've already deployed the lambda functions in aws using terragrunt and made some changes in the code that need to roll out, and then we face above issue.
- We use aws SSO
- S3 bucket :  tfstate-devops-123456789-eu-west-1 is already existed and has enough permissions as well, Cleared the terragrunt cache and the issue still persist, could someone help how to overcome this?

Terragrunt version: v0.38.7

Thanks in advance.

---

<ins datetime="2022-12-07T16:45:01Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109701">Tracked in ticket #109701</a></p>
</ins>

**Comment by gitsstewart** [comment]
Without knowing what led up to this question or what you're trying to accomplish before you hit these errors it's tough to help. Could you provide the relevant sections of your config, and what is happening before running terragrunt init?
Could you provide more details about the context surrounding this error? As an example, see what [this person provided](https://github.com/gruntwork-io/knowledge-base/discussions/355) in their question.

**Reply by skkc2** [reply]
We're trying to update the lambda function in aws, we've already deployed the lambda functions in aws using terragrunt and made some changes in the code that need to roll out, and then we face above issue.
We use AWS SSO

**Comment by denis256** [comment]
Hi,
will be helpful if Terragrunt version will be mentioned and also provided output from running with `--terragrunt-log-level debug` flag

**Reply by skkc2** [reply]
Terragrunt version: v0.38.7
After running the $ terragrunt terragrunt-log-level debug

> Remote state S3 bucket tfstate-devops-123456789-eu-west-1 does not exist or you don't have permissions to access it. Would you like Terragrunt to create it? (y/n) n
> level=error msg=remote state S3 bucket tfstate-devops-123456789-eu-west-1 does not exist or you don't have permissions to access it
>level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1

**Reply by denis256** [reply]
hi,
I suspect it may be an issue with assuming the right IAM role, can be attempted to run terragrunt through `aws-valut` which supports handling of SSO 

https://github.com/99designs/aws-vault/blob/master/USAGE.md#aws-single-sign-on-aws-sso
